troposphere/static/js/components/modals/instance/InstanceLaunchWizardModal.jsx

```diff
@@ -628,6 +628,36 @@ export default React.createClass({
             });
         }
 
+        if (providerSizeList && imageVersion) {
```

Comment:
This is a good point, you could call this a "SLAP" violation. 

SLAP stands for single-level of abstraction principle. This is something that Uncle Bob Martin talks about, also Neal Ford - and you can see it discussed here:
- http://www.markhneedham.com/blog/2009/06/12/coding-single-level-of-abstraction-principle/

(it is also mentioned on the internal CyVerse Core Service [wiki](https://pods.iplantcollaborative.org/wiki/display/csmgmt/Good+Programming+Practice))

Thank you for mentioning this.